numpy-stubs/__init__.pyi

```diff
@@ -462,34 +466,60 @@ class uint32(unsignedinteger):
 class uint64(unsignedinteger):
     def __init__(self, value: SupportsInt = ...) -> None: ...
 
-class inexact(number): ...
-class floating(inexact, _real_generic): ...
-class float16(floating): ...
-class float32(floating): ...
-class float64(floating): ...
+class inexact(number, metaclass=ABCMeta): ...
+class floating(inexact, _real_generic, metaclass=ABCMeta): ...
 
-class complexfloating(inexact):
-    def __init__(
-        self, value: Union[SupportsInt, SupportsFloat, SupportsComplex, complex] = ...
-    ) -> None: ...
+class float16(floating):
+    def __init__(self, value: SupportsFloat = ...) -> None: ...
+
+class float32(floating):
+    def __init__(self, value: SupportsFloat = ...) -> None: ...
+
+class float64(floating):
+    def __init__(self, value: SupportsFloat = ...) -> None: ...
+
+class complexfloating(inexact, metaclass=ABCMeta): ...
 
 class complex64(complexfloating):
+    def __init__(
+        self, value: Union[SupportsInt, SupportsFloat, SupportsComplex] = ...
+    ) -> None: ...
     @property
     def real(self) -> float32: ...
     @property
     def imag(self) -> float32: ...
 
 class complex128(complexfloating):
+    def __init__(
+        self, value: Union[SupportsInt, SupportsFloat, SupportsComplex] = ...
+    ) -> None: ...
     @property
     def real(self) -> float64: ...
     @property
     def imag(self) -> float64: ...
 
-class flexible(_real_generic): ...
-class void(flexible): ...
-class character(_real_generic): ...
-class bytes_(character): ...
-class str_(character): ...
+class flexible(_real_generic, metaclass=ABCMeta): ...
+
+class void(flexible):
+    def __init__(self, value: Union[int, integer, bool_, bytes, bytes_]): ...
+
+class character(_real_generic, metaclass=ABCMeta): ...
+
+class bytes_(character):
```

Comment:
So contrary to its ``bytes.__init__()`` counterpart in typeshed ([ref](https://github.com/python/typeshed/blob/e5dcf06c9e68969fc28fb0a869e281f15cf4a1dd/stdlib/2and3/builtins.pyi#L559-L570)), it seems that ``np.bytes_`` can truly take any object as argument.

```python
>>> import numpy as np

>>> np.bytes_(0)
b''

>>> np.bytes_(None)
b'None'  # Equivalent to np.bytes_(str(None))

>>> class Test: 
...     pass

>>> np.bytes_(Test)
b"<class '__main__.Test'>"
>>> np.bytes_(Test())
b'<__main__.Test object at ...>'
```